services/galley/test/integration/API/Teams.hs

```diff
@@ -115,6 +116,48 @@ testCreateTeamWithMembers g b c = do
         e^.eventTeam @?= (team^.teamId)
         e^.eventData @?= Just (EdTeamCreate team)
 
+testCreateOne2OneWithMembers :: Galley -> Brig -> Cannon -> Maybe Aws.Env -> Http ()
+testCreateOne2OneWithMembers g b c a = do
+    -- Negative tests first
+    ensureNoOne2OneUnlessSameBindingTeam
+    
+    -- Test with binding teams
+    owner <- Util.randomUser b
+    tid   <- Util.createTeamInternal g "foo" owner
+    assertQueue a tCreate
+    let p1 = Util.symmPermissions [CreateConversation]
+    mem1 <- flip newTeamMember p1 <$> Util.randomUser b
+
+    WS.bracketR c (mem1^.userId) $ \wsMem1 -> do
+        Util.addTeamMemberInternal g tid mem1
+        checkTeamMemberJoin tid (mem1^.userId) wsMem1
+        WS.assertNoEvent timeout [wsMem1]
+
+    Util.createO2OTeamConv g owner (mem1^.userId) Nothing tid !!! const 201 === statusCode
+  where
+    ensureNoOne2OneUnlessSameBindingTeam = do
```

Comment:
As this test seems independent, simply make it an independent test rather than a local function?

Reply:
Yes, seems like a better idea